### DISCUSSION TITLE
Publish a fork of the Bitwarden web browsers extension to allow usage of multiple logins

### DISCUSSION THREAD

**Discussion by MurzNN** [dani-garcia/vaultwarden#4231] [Ideas]
Bitwarden introduces a long-awaited multi-account feature for browser extensions here https://community.bitwarden.com/t/account-switching-browser-support-wip/716 and promises to release it in January 2024.

But it actually doesn’t make usage of two Bitwarden logins a lot easier!

Because users will still have to manually switch between their two accounts many times per day to log in to different websites using their first or second account (or even third).

For example, a user has a personal account and a corporate account, or two corporate accounts (if they work for two companies). And every time to log in to a corporate account on a website - they will need to open the Bitwarden extension and switch to the corporate account, then after a minute - switch back to their personal account to log in to another website using their personal account, and repeat this ritual dozens of times per day!

The easiest solution to resolve this problem and make use of two Bitwarden accounts simultaneously.

This will be especially convenient for Vaultwarden users because they very often have multiple instances of the Vaultwarden and have to suffer from manual switching between accounts multiple times per day.

I and other users asking Bitwarden to do this for already long time and many times, but Bitwarden company just ignoring these requests.

To resolve this unpleasantness, it will be great to publish the secondary extension to the Firefox and Chrome stores from the Vaultwarden project!

So, just make a fork of the https://github.com/bitwarden/clients/tree/main/apps/browser, replace the name and logos to the Vaultwarden, publish to browsers' stores, and that's it!

And publishing a fork of the Bitwarden browser extension from the pretty good known brand "Vaultwarden" will gain much more trust, than from just a regular random Github user.

What do you think about this idea?

**Comment by stefan0xC** [comment]
I don't think that's something we have the resources for (I mean the fork would have to be different enough so it could be installed besides the Bitwarden extension and also not interfere, we'd have to maintain the fork and also publishing an extension to different browsers is also quite the undertaking as far as I know). I'd rather recommend you wait till the account switching feature works (and then you can evaluate if it really is as cumbersome as you think). In the meantime you can also use two different browsers (e.g. Firefox for Private, Chrome for your corporate account). Also Firefox allows to setup [different profiles](https://support.mozilla.org/en-US/kb/profile-manager-create-remove-switch-firefox-profiles) where the extensions would be independent accessible of each other, which might be another alternative I recommend.

**Reply by MurzNN** [reply]
Thanks for the explanation! I already asked about how the multi-account feature will work from Bitwarden developers, and they confirmed that I will have to switch between accounts manually dozens of times per day.

So, I posted the same feature request to Birwarden: https://community.bitwarden.com/t/publish-a-secondary-copy-of-the-bitwarden-browser-extension-to-allow-use-two-logins-at-the-same-time/61966 but the hopes that they will do it are almost zero, from my experience.

And yes, I understand that maintaining the fork is an additional routine, but I think this is actually not a lot of work, because we can automate updating the fork and publishing new releases using Github actions:
- https://github.com/marketplace/actions/firefox-addon-action
- https://github.com/marketplace/actions/chrome-extension-upload-publish

So, maybe there will be any volunteers who will do this automation?

**Comment by MurzNN** [comment]
And I even found here https://github.com/unixtools/custom-bitwarden a helper to build a copy of the Bitwarden extension from @nneul - maybe he would be happy to help with this?

**Comment by MurzNN** [comment]
Also, found already released forks here https://github.com/bitwarden/clients/issues/1058#issuecomment-591939724 from @LaurentSGL and here https://github.com/bitwarden/clients/issues/1058#issuecomment-1086927430 from @levihb

**Comment by MurzNN** [comment]
And publishing a fork of the Bitwarden browser extension from the pretty good known brand "Vaultwarden" will gain much more trust, than from just a regular random Github user.

**Comment by BlackDex** [comment]
As @stefan0xC already mentioned, no resources to do that. Also, my knowledge of the language and tools used to build and maintain that is sub-par. We have enough to do and stuff laying around for Vaultwarden it self, let alone add something new into the mix and keeping it up-to-date also.

**Comment by MurzNN** [comment]
I understand about lack of the resources. But if any volunteers are found who are ready to take on this work, will you allow them to make this from the Vaultwarden name?

**Reply by stefan0xC** [reply]
Unless we absolutely have to I am against forking the extension whether it's done by us or in our name - if others have already done so, you can of course use that but I'd rather not that we promote a third-party extension since then it becomes this project's reponsibility to check whether the extension is safe to endorse.

I mean, I also don't understand the need for this browser extension to be called Vaultwarden. What exactly would be the tie-in to this project if it's not maintained by us and could just as easily be used for a self-hosted Bitwarden instance? The idea of your fork is to allow the usage of a second login to a Bitwarden-compatible instance which I'd argue is much easier accomplished by other means (e.g. using two browsers with the official Bitwarden extension).

**Comment by Pieterv24** [comment]
Would it be an option to contribute this this to the original extension? Instead of creating a fork.

Or has Bitwarden explicitly stated they will not support this usecase?

**Comment by nneul** [comment]
They haven't seemed to have any interest in publishing a second extension, and the current 'switching' function that is coming has taken 2 years if I remember right and still isn't available.

The main reason I did my workaround for this was that I have 4 separate vaultwarden instances that I bounce between, often in the same browser -- and that's in addition to having multiple browser profiles. Just having multiple isn't quite enough though, since then you get multiple identical bitwarden icons on the browser. The build script that I use in that repo creates multiple with distinct icons and menu names so that I have 'Bitwarden' for the store-installed one, and then 'Bitwarden - CompanyA' and 'Bitwarden - CompanyB' for the others. 

Certainly embedding that capability right into the extension would be great, but that's outside my skill area.

**Reply by Pieterv24** [reply]
I think in that case I might have some good, and some bad news.
The good news is: (as far as I've seen), the multi account funcionality has been added in 2024.1.0.
The bad news is, that as far as I know, you still have to select the account every time.
So you have to switch to the correct vault every time.

**Comment by Gerardv514** [comment]
How many users would actually benefit from the proposal here? Personally I would think this is a lot to maintain and would rather have efforts be put forth on the main project.

**Comment by elikoga** [comment]
Perhaps there is renewed interest now that Bitwarden has decided to roll out a terrible UI overhaul that does nothing except lower information density and making it difficult to find all the buttons. ~~I think they have been sabotaged to commit suicide or something?~~

https://community.bitwarden.com/t/option-to-revert-browser-extension-to-previous-ui/77395/4

Can't even see my comment because it's waiting for admin approval.

**Comment by MurzNN** [comment]
Yeah, the new design is terrible, UX is very bad! Started to spam them by feedback on how to make it less ugly 🙂
